### ISSUE TITLE
Connect to Zcash mainnet or testnet

### ISSUE THREAD

**Issue by s-tikhomirov** [s-tikhomirov/bcclient#10]
Currently: connection established, Version sent, no answer received (connection closed). Full node (zcashd) works.

**Comment by s-tikhomirov** [comment]
* 10.186.24.37 (port 8233)
* [list of peers](https://www.coinexchange.io/network/peers/ZEC)

**Comment by s-tikhomirov** [comment]
`sudo tcpdump tcp port <PORT>` -- sniff network traffic.

Bitcoin mainnet (success):
```
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on enx0050b612e5c5, link-type EN10MB (Ethernet), capture size 262144 bytes
19:24:15.322113 IP guilvinec.uni.lux.40002 > 46.101.217.211.8333: Flags [S], seq 4210887977, win 29200, options [mss 1460,sackOK,TS val 3715069696 ecr 0,nop,wscale 7], length 0
19:24:15.331935 IP 46.101.217.211.8333 > guilvinec.uni.lux.40002: Flags [S.], seq 1192973603, ack 4210887978, win 28960, options [mss 1460,sackOK,TS val 2969831199 ecr 3715069696,nop,wscale 6], length 0
19:24:15.331959 IP guilvinec.uni.lux.40002 > 46.101.217.211.8333: Flags [.], ack 1, win 229, options [nop,nop,TS val 3715069706 ecr 2969831199], length 0
19:24:15.332220 IP guilvinec.uni.lux.40002 > 46.101.217.211.8333: Flags [P.], seq 1:125, ack 1, win 229, options [nop,nop,TS val 3715069706 ecr 2969831199], length 124
19:24:15.338859 IP 46.101.217.211.8333 > guilvinec.uni.lux.40002: Flags [.], ack 125, win 453, options [nop,nop,TS val 2969831207 ecr 3715069706], length 0
19:24:15.338876 IP guilvinec.uni.lux.40002 > 46.101.217.211.8333: Flags [P.], seq 125:149, ack 1, win 229, options [nop,nop,TS val 3715069713 ecr 2969831207], length 24
19:24:15.345526 IP 46.101.217.211.8333 > guilvinec.uni.lux.40002: Flags [.], ack 149, win 453, options [nop,nop,TS val 2969831214 ecr 3715069713], length 0
19:24:15.952060 IP 46.101.217.211.8333 > guilvinec.uni.lux.40002: Flags [P.], seq 1:127, ack 149, win 453, options [nop,nop,TS val 2969831820 ecr 3715069713], length 126
19:24:15.952101 IP guilvinec.uni.lux.40002 > 46.101.217.211.8333: Flags [.], ack 127, win 229, options [nop,nop,TS val 3715070326 ecr 2969831820], length 0
19:24:15.952109 IP 46.101.217.211.8333 > guilvinec.uni.lux.40002: Flags [P.], seq 127:151, ack 149, win 453, options [nop,nop,TS val 2969831820 ecr 3715069713], length 24
19:24:15.952116 IP guilvinec.uni.lux.40002 > 46.101.217.211.8333: Flags [.], ack 151, win 229, options [nop,nop,TS val 3715070326 ecr 2969831820], length 0
19:24:15.952663 IP guilvinec.uni.lux.40002 > 46.101.217.211.8333: Flags [P.], seq 149:173, ack 151, win 229, options [nop,nop,TS val 3715070327 ecr 2969831820], length 24
19:24:15.952860 IP guilvinec.uni.lux.40002 > 46.101.217.211.8333: Flags [F.], seq 173, ack 151, win 229, options [nop,nop,TS val 3715070327 ecr 2969831820], length 0
19:24:15.960256 IP 46.101.217.211.8333 > guilvinec.uni.lux.40002: Flags [.], ack 173, win 453, options [nop,nop,TS val 2969831828 ecr 3715070327], length 0
19:24:15.960561 IP 46.101.217.211.8333 > guilvinec.uni.lux.40002: Flags [F.], seq 151, ack 174, win 453, options [nop,nop,TS val 2969831829 ecr 3715070327], length 0
19:24:15.960594 IP guilvinec.uni.lux.40002 > 46.101.217.211.8333: Flags [.], ack 152, win 229, options [nop,nop,TS val 3715070335 ecr 2969831829], length 0
```

Zcash mainnet (failure):
```
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on enx0050b612e5c5, link-type EN10MB (Ethernet), capture size 262144 bytes
19:17:50.215916 IP guilvinec.uni.lux.35248 > detector4.z.cash.8233: Flags [S], seq 1866882811, win 29200, options [mss 1460,sackOK,TS val 1193098061 ecr 0,nop,wscale 7], length 0
19:17:50.224115 IP detector4.z.cash.8233 > guilvinec.uni.lux.35248: Flags [S.], seq 2265170014, ack 1866882812, win 28960, options [mss 1460,sackOK,TS val 820790631 ecr 1193098061,nop,wscale 7], length 0
19:17:50.224174 IP guilvinec.uni.lux.35248 > detector4.z.cash.8233: Flags [.], ack 1, win 229, options [nop,nop,TS val 1193098070 ecr 820790631], length 0
19:17:50.224817 IP guilvinec.uni.lux.35248 > detector4.z.cash.8233: Flags [P.], seq 1:125, ack 1, win 229, options [nop,nop,TS val 1193098070 ecr 820790631], length 124
19:17:50.231521 IP detector4.z.cash.8233 > guilvinec.uni.lux.35248: Flags [.], ack 125, win 227, options [nop,nop,TS val 820790633 ecr 1193098070], length 0
19:17:50.231580 IP guilvinec.uni.lux.35248 > detector4.z.cash.8233: Flags [P.], seq 125:149, ack 1, win 229, options [nop,nop,TS val 1193098077 ecr 820790633], length 24
19:17:50.238452 IP detector4.z.cash.8233 > guilvinec.uni.lux.35248: Flags [.], ack 149, win 227, options [nop,nop,TS val 820790635 ecr 1193098077], length 0
```

**Comment by s-tikhomirov** [comment]
Currently: can not connect in any way. Testnet and mainnet nodes drop connection even if full nodes connect. Maybe reference implementation drops connections depending on version or smth?

Pattern ([tcpdump](http://amits-notes.readthedocs.io/en/latest/networking/tcpdump.html)):

[S] - [S.] - [.] [P.] - [.] - [P.] - **[F.] - [F.] - [R]** / **[R.] [R]**

**Comment by s-tikhomirov** [comment]
Hypothesis: expected message structure differs between Bitcoin and Zcash, remote Zcash nodes discard my messages as invalid.

[Bitcoin protocol documentation](https://en.bitcoin.it/wiki/Protocol_documentation)

**Comment by s-tikhomirov** [comment]
Fixed: added magic numbers where necessary (a221e396f695d0b9d5d039a93fe4c73c6c4f9f81), connected to mainnet.
